pkg/cache/cache.go

```diff
@@ -126,7 +126,7 @@ func defaultOpts(config *rest.Config, opts Options) (Options, error) {
 	// Construct a new Mapper if unset
 	if opts.Mapper == nil {
 		var err error
-		opts.Mapper, err = apiutil.NewDiscoveryRESTMapper(config)
+		opts.Mapper, err = apiutil.NewDynamicRESTMapper(config)
```

Comment:
This is a good change but should be its own PR as it doesn't really have anything to do with this PR

Reply:
Fair enough, I will move this into a separate PR. 👍🏻